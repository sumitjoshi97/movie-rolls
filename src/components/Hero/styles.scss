@import "../../sass/variables";
@import "../../sass/mixin";

.hero {
  width: 100%;
  height: 100vh;
  line-height: 3rem;
  background-repeat: no-repeat;
  background-position: center;
  background-size: cover;
  display: flex;
  align-items: center;

  a {
    text-decoration: none;
    color: $color-white;
  }
  &__info {
    position: relative;
    padding: 0 8rem;
    display: flex;
    flex-direction: row;
    @include respond(phone) {
      padding: 0 4rem;
      flex-direction: column;
    }

    &__img {
      width: 20rem;
      height: 30rem;
      border-radius: 2px;

      @include respond(phone) {
        margin-bottom: 4rem;
      }
    }
    &__text {
      padding-left: 2rem;
      width: 70rem;
      position: relative;

      @include respond(phone) {
        padding-left: 0;
        width: auto;
      }

      &__header {
        text-transform: capitalize;
      }

      &__rating {
        display: flex;
        align-items: center;
        padding-bottom: 1rem;
      }

      &__language {
      }
      &__adult {
      }

      &__date {
      }
      &__type {
        text-transform: capitalize;

        & > * {
          padding-right: 1rem;
          color: #888888;
        }

        &__adult {
          border: 1px solid #888888;
          border-radius: 4px;
          padding: 0.5rem;
          margin-right: 1rem;
        }
      }

      &__summary {
        color: #ddd;
        padding-bottom: 1rem;
        position: relative;
        width: 60rem;

        @include respond(phone) {
          width: auto;
        }
      }
      &__action-btns {
        display: flex;
        flex-direction: row;

        & > * {
          padding: 1rem;
          font-size: 1.4rem;
          position: relative;
          text-transform: capitalize;
          background: transparent;
          color: $color-white;
          border: 1px solid $color-white;
          border-radius: 4px;
          cursor: pointer;
          &:active,
          &:focus {
            outline: none;
          }
        }

        &__favorite {
          margin-right: 1rem;
        }
        &__watch-later {
          margin-left: 1rem;
        }
      }
    }
  }
}
