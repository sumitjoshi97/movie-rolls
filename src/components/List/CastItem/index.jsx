import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'

export default class CastItem extends PureComponent {
  static propTypes = {
    poster: PropTypes.string,
    name: PropTypes.string.isRequired,
    role: PropTypes.string
  }

  render() {
    const { poster, name, role } = this.props

    return (
      <div className='list-item'>
        <img
          src={`https://image.tmdb.org/t/p/w185/${poster}`}
          alt={name}
          className='list-item__img'
        />
        <div className='list__item__name'>{name}</div>
        <div className='list__item__name__sub'>{role}</div>
      </div>
    )
  }
}
