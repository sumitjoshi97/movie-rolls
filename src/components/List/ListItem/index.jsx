import React, { PureComponent } from 'react'
import { Link } from 'react-router-dom'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'

import * as actions from '../../../actions'

class ListItem extends PureComponent {
  static propTypes = {
    pageId: PropTypes.number.isRequired,
    type: PropTypes.string.isRequired,
    poster: PropTypes.string,
    name: PropTypes.string.isRequired,
    setType: PropTypes.func.isRequired
  }

  render() {
    const { pageId, type, poster, name, setType } = this.props

    return (
      <div className='list-item'>
        <Link to={`/details/${pageId}`} onClick={() => setType(type)}>
          <img
            src={`https://image.tmdb.org/t/p/w185/${poster}`}
            alt={name}
            className='list-item__img'
          />
          <div className='list__item__name'>{name}</div>
        </Link>
        )
      </div>
    )
  }
}

const mapDispatchToProps = dispatch => ({
  setType: type => dispatch(actions.setItemType(type))
})

export default connect(
  null,
  mapDispatchToProps
)(ListItem)
