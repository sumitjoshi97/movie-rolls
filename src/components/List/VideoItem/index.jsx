import React from 'react'
import PropTypes from 'prop-types'

const VideoItem = ({ video, name, category }) => (
  <div className='list-item'>
    <iframe
      className='list-item__video'
      title='1'
      width='280'
      src={`https://www.youtube.com/embed/${video}`}
    />
    <div className='list-item__name'>{name}</div>
    <div className='list-item__name__sub'>{category}</div>
  </div>
)

VideoItem.propTypes = {
  video: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  category: PropTypes.string
}

export default VideoItem
