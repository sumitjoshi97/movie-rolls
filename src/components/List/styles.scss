.list {
  padding: 6rem 6rem 0 6rem;
  background-color: #1a1a1a;
  &__header {
    text-transform: capitalize;
    padding-bottom: 1rem;
  }
}

.list-item {
  display: inline-block;
  padding: 2rem;
  &__img {
    height: 18rem;
    background: #000;
    width: 14rem;
  }

  &__video {
    display: inline-block;
    height: 18rem;
    outline: none;
    border: none;
  }

  &__name {
  }
}
