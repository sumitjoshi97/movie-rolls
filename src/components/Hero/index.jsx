import React from 'react'
import PropTypes from 'prop-types'
import RatingStars from '../RatingStars'
import './styles.scss'

const Hero = ({
  title,
  poster,
  backdrop,
  rating,
  type,
  summary,
  adult,
  year
}) => {
  const heroBackdrop = {
    backgroundImage: `linear-gradient(to bottom, rgba(0,0,0,0.8), rgba(0,0,0,0.9)), url(https://image.tmdb.org/t/p/original${backdrop})`
  }

  return (
    <div className='hero' style={heroBackdrop}>
      <div className='hero__info'>
        <img
          src={`https://image.tmdb.org/t/p/w185/${poster}`}
          alt=''
          className='hero__info__img'
        />
        <div className='hero__info__text'>
          <h1 className='hero__info__text__header'>{title}</h1>
          <div className='hero__info__text__rating'>
            {rating}
            <RatingStars stars={rating} />
          </div>
          <div className='hero__info__text__type'>
            <span>{type}</span>
            <span className='hero__info__text__adult'>
              {adult ? '18+' : ''}
            </span>

            <span className='hero__info__text__year'>{year}</span>
          </div>

          <p className='hero__info__text__summary'>{summary}</p>
          <button className='hero__info__text__favorite-btn'>
            add to favorite
          </button>
        </div>
      </div>
    </div>
  )
}

Hero.propTypes = {
  title: PropTypes.string.isRequired,
  poster: PropTypes.string,
  backdrop: PropTypes.string,
  rating: PropTypes.number,
  summary: PropTypes.string,
  adult: PropTypes.bool,
  year: PropTypes.string
}

export default Hero
