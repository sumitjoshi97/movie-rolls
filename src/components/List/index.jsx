import React, { PureComponent } from 'react'
import Slider from 'react-slick'

import ListItem from './ListItem'
import CastItem from './CastItem'
import VideoItem from './VideoItem'
import NextArrow from '../Arrows/Next'
import PrevArrow from '../Arrows/Prev'

import './styles.scss'

export default class List extends PureComponent {
  renderList = () => {
    const { type, items } = this.props

    switch (type) {
      case 'cast':
        return items.map(item => (
          <CastItem
            key={item.id}
            name={item.name}
            poster={item.profile_path}
            role={item.character}
          />
        ))

      case 'video':
        return items.map(item => (
          <VideoItem
            key={item.id}
            name={item.name}
            video={item.key}
            category={item.type}
          />
        ))

      //for movies / tv shows
      default:
        return items.map(item => (
          <ListItem
            key={item.id}
            type={type}
            pageId={item.id}
            name={item.title}
            rating={item.vote_average}
            poster={item.poster_path}
          />
        ))
    }
  }

  render() {
    const sliderOptions = {
      arrows: true,
      dots: false,
      speed: 400,
      slidesToShow: this.props.type === 'video' ? 4 : 7,
      swipeToSlide: true,
      infinite: false,
      nextArrow: <NextArrow />,
      prevArrow: <PrevArrow />
    }
    const { name } = this.props
    return (
      <div className='list'>
        <h2 className='list__header'>{name}</h2>
        <Slider {...sliderOptions}>{this.renderList()}</Slider>
      </div>
    )
  }
}
