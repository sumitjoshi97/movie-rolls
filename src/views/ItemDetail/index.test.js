import React from 'react'
import { shallow } from 'enzyme'

import { ItemDetail } from './index'
import Hero from '../../components/Hero'
import List from '../../components/List'

describe('Item Details unit tests', () => {
  const match = {
    params: {
      itemId: '01'
    }
  }
  const type = 'movie'
  const item = {
    id: '01',
    title: 'test',
    poster_path: null,
    backdrop_path: null,
    vote_average: 4,
    type: 'movie',
    summary: 'test summary',
    adult: false,
    release_date: '2019-15-1'
  }
  const credits = { cast: [{ id: '01' }] }
  const videos = null
  const similar = {
    results: [{ id: '02' }]
  }
  const fetchQueryItem = jest.fn()
  const fetchQueryItemDetails = jest.fn()
  const fetchQueryItemVideos = jest.fn()
  const fetchSimilarItems = jest.fn()
  const setItemClear = jest.fn()

  const props = {
    match,
    type,
    item,
    credits,
    videos,
    similar,
    fetchQueryItem,
    fetchQueryItemDetails,
    fetchQueryItemVideos,
    fetchSimilarItems,
    setItemClear
  }

  const itemDetailWrapper = shallow(<ItemDetail {...props} />)

  test('should render properly', () => {
    expect(itemDetailWrapper).toMatchSnapshot()
  })

  test('should render hero of type `movie`', () => {
    expect(itemDetailWrapper.find({ type: 'movie' }).find(Hero)).toBeTruthy()
  })

  test('should render list of cast', () => {
    expect(itemDetailWrapper.find({ type: 'cast' }).find(List)).toBeTruthy()
  })

  test('should call fetchQueryItem fn', () => {
    expect(fetchQueryItem).toHaveBeenCalledWith(type, match.params.itemId)
  })
  test('should call fetchQueryItemDetails fn', () => {
    expect(fetchQueryItemDetails).toHaveBeenCalledWith(
      type,
      match.params.itemId
    )
  })
  test('should call fetchQueryItemVideos fn', () => {
    expect(fetchQueryItemVideos).toHaveBeenCalledWith(type, match.params.itemId)
  })
  test('should call fetchSimilar fn', () => {
    expect(fetchSimilarItems).toHaveBeenCalledWith(type, match.params.itemId)
  })
})
