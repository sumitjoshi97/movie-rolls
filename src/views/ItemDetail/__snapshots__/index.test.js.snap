// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item Details unit tests should render properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ItemDetail
    credits={
      Object {
        "cast": Array [
          Object {
            "id": "01",
          },
        ],
      }
    }
    fetchQueryItem={
      [MockFunction] {
        "calls": Array [
          Array [
            "movie",
            "01",
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    fetchQueryItemDetails={
      [MockFunction] {
        "calls": Array [
          Array [
            "movie",
            "01",
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    fetchQueryItemVideos={
      [MockFunction] {
        "calls": Array [
          Array [
            "movie",
            "01",
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    fetchSimilarItems={
      [MockFunction] {
        "calls": Array [
          Array [
            "movie",
            "01",
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    item={
      Object {
        "adult": false,
        "backdrop_path": null,
        "id": "01",
        "poster_path": null,
        "release_date": "2019-15-1",
        "summary": "test summary",
        "title": "test",
        "type": "movie",
        "vote_average": 4,
      }
    }
    match={
      Object {
        "params": Object {
          "itemId": "01",
        },
      }
    }
    setItemClear={[MockFunction]}
    similar={
      Object {
        "results": Array [
          Object {
            "id": "02",
          },
        ],
      }
    }
    type="movie"
    videos={null}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="item-details__hero"
        >
          <Connect(withRouter(Hero))
            adult={false}
            backdrop={null}
            pageId="01"
            poster={null}
            rating={4}
            title="test"
            type="movie"
            year="2019"
          />
          <Nav
            handleNav={[Function]}
            options={
              Array [
                Object {
                  "name": "overview",
                  "type": "overview",
                },
                Object {
                  "name": "more like this",
                  "type": "similar",
                },
              ]
            }
          />
        </div>,
        <div
          className="item-details__lists"
        >
          <React.Fragment>
            <List
              items={
                Array [
                  Object {
                    "id": "01",
                  },
                ]
              }
              name="Cast"
              type="cast"
            />
          </React.Fragment>
        </div>,
      ],
      "className": "item-details",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Connect(withRouter(Hero))
              adult={false}
              backdrop={null}
              pageId="01"
              poster={null}
              rating={4}
              title="test"
              type="movie"
              year="2019"
            />,
            <Nav
              handleNav={[Function]}
              options={
                Array [
                  Object {
                    "name": "overview",
                    "type": "overview",
                  },
                  Object {
                    "name": "more like this",
                    "type": "similar",
                  },
                ]
              }
            />,
          ],
          "className": "item-details__hero",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "01",
            "nodeType": "class",
            "props": Object {
              "adult": false,
              "backdrop": null,
              "pageId": "01",
              "poster": null,
              "rating": 4,
              "summary": undefined,
              "title": "test",
              "type": "movie",
              "year": "2019",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "handleNav": [Function],
              "options": Array [
                Object {
                  "name": "overview",
                  "type": "overview",
                },
                Object {
                  "name": "more like this",
                  "type": "similar",
                },
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <React.Fragment>
            <List
              items={
                Array [
                  Object {
                    "id": "01",
                  },
                ]
              }
              name="Cast"
              type="cast"
            />
          </React.Fragment>,
          "className": "item-details__lists",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <List
                items={
                  Array [
                    Object {
                      "id": "01",
                    },
                  ]
                }
                name="Cast"
                type="cast"
              />,
              null,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "items": Array [
                  Object {
                    "id": "01",
                  },
                ],
                "name": "Cast",
                "type": "cast",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            null,
          ],
          "type": Symbol(react.fragment),
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="item-details__hero"
          >
            <Connect(withRouter(Hero))
              adult={false}
              backdrop={null}
              pageId="01"
              poster={null}
              rating={4}
              title="test"
              type="movie"
              year="2019"
            />
            <Nav
              handleNav={[Function]}
              options={
                Array [
                  Object {
                    "name": "overview",
                    "type": "overview",
                  },
                  Object {
                    "name": "more like this",
                    "type": "similar",
                  },
                ]
              }
            />
          </div>,
          <div
            className="item-details__lists"
          >
            <React.Fragment>
              <List
                items={
                  Array [
                    Object {
                      "id": "01",
                    },
                  ]
                }
                name="Cast"
                type="cast"
              />
            </React.Fragment>
          </div>,
        ],
        "className": "item-details",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Connect(withRouter(Hero))
                adult={false}
                backdrop={null}
                pageId="01"
                poster={null}
                rating={4}
                title="test"
                type="movie"
                year="2019"
              />,
              <Nav
                handleNav={[Function]}
                options={
                  Array [
                    Object {
                      "name": "overview",
                      "type": "overview",
                    },
                    Object {
                      "name": "more like this",
                      "type": "similar",
                    },
                  ]
                }
              />,
            ],
            "className": "item-details__hero",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "01",
              "nodeType": "class",
              "props": Object {
                "adult": false,
                "backdrop": null,
                "pageId": "01",
                "poster": null,
                "rating": 4,
                "summary": undefined,
                "title": "test",
                "type": "movie",
                "year": "2019",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "handleNav": [Function],
                "options": Array [
                  Object {
                    "name": "overview",
                    "type": "overview",
                  },
                  Object {
                    "name": "more like this",
                    "type": "similar",
                  },
                ],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <React.Fragment>
              <List
                items={
                  Array [
                    Object {
                      "id": "01",
                    },
                  ]
                }
                name="Cast"
                type="cast"
              />
            </React.Fragment>,
            "className": "item-details__lists",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <List
                  items={
                    Array [
                      Object {
                        "id": "01",
                      },
                    ]
                  }
                  name="Cast"
                  type="cast"
                />,
                null,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "items": Array [
                    Object {
                      "id": "01",
                    },
                  ],
                  "name": "Cast",
                  "type": "cast",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              null,
            ],
            "type": Symbol(react.fragment),
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
